packages/parser/tsconfig.json

```diff
@@ -0,0 +1,13 @@
+{
+  "extends": "../../tsconfig.json",
+  "compilerOptions": {
+    "rootDir": "./src",
+    "outDir": "./dist",
+    // TODO: figure out how to remove these configs
+    "moduleResolution": "node",
```

Comment:
Is this still required after adding @chevrotain/types to dependencies?

Reply:
> It's fine that there's _some_ common stuff shared. Looking at `MermiadTokenBuilder`, it's fine to have it shared. The goal is to not have to load 10 different grammars to render a flowchart.

@sidharthv96 It would be fine for common stuff to be there, but I'm talking about other stuff. For example, the section title in the `timeline` diagram has a custom match with the `section` keyword trimmed out, then more `switch .. case` need to added (some rule might have conflict since they have same name, although most of them don't until now).

> The thing with test cases is that most of the heavy lifting is done by the image snapshot tests, not unit tests. It's not ideal, but that's what we have now. You can run them locally first in develop to generate the reference images, and run it again to compare the outputs if you want to do it locally.
> 
> Now that you have 2 diagrams ready, I'd suggest giving it a shot to get it working inside the webpage, to actually render diagrams.

Some stuff hasn't been handled yet, so I'm not sure if I should go rendering now, and it takes more time than unit testing to validate, so I would stick with it for now.

Reply:
No. I'm gonna remove it now, but the below options appears that can't be removed, it throws exceptions when building with `tsc`.

But there is an issue that prevent building even after adding those options:

```
../../node_modules/.pnpm/langium@1.2.0/node_modules/langium/src/dependency-injection.ts:92:113 - error TS2554: Expected 0-1 arguments, but got 2.

92             throw new Error('Construction failure. Please make sure that your dependencies are constructable.', {cause: obj[prop]});
                                                                                                                   ~~~~~~~~~~~~~~~~~~


Found 1 error in ../../node_modules/.pnpm/langium@1.2.0/node_modules/langium/src/dependency-injection.ts:92

 ELIFECYCLE  Command failed with exit code 2.
```

if @msujew can us a solution that's would be great!

Reply:
Hm, it seems like the `options` argument for the `Error` constructor is only supported in `ES2022`. I'm wondering though why your code tries to compile our TypeScript code (that we only ship in our package for debugging purposes). Shouldn't it just use our generated JS code?

Reply:
> I'm wondering though why your code tries to compile our TypeScript code (that we only ship in our package for debugging purposes). Shouldn't it just use our generated JS code?

I'm not sure how to, tbh. If you could create sample code showing this, I'd be grateful!